### ISSUE TITLE
Running eval.py on custom dataset and mAP = 0

### ISSUE THREAD

**Issue by cattyhubby** [amdegroot/ssd.pytorch#443]
Dear all.
I am running SSD300 on a custom chest X-ray dataset preprocessed as VOC format, so far train.py, test.py and demo.py works well while for the eval.py, I am getting the results as follows,
VOC07 metric? Yes
Reading annotation for 1/264
Reading annotation for 101/264
Reading annotation for 201/264
Saving cached annotations to /home/yuyang/data/VOCdevkit/VOCNIH14/annotations_cache/annots.pkl
AP for atelectasis = 0.0000
AP for cardiomegaly = 0.0000
AP for effusion = 0.0000
AP for infiltrate = 0.0000
AP for mass = 0.0000
AP for nodule = 0.0000
AP for pneumonia = 0.0000
AP for pneumothorax = 0.0000
Mean AP = 0.0000
~~~~~~~~
Results:
0.000
0.000
0.000
0.000
0.000
0.000
0.000
0.000
0.000
~~~~~~~~

Does anyone know how may I solve the issue? Thanks.

**Comment by AtriSaxena** [comment]
You actually need to do some changes in eval.py file. 
You need to change the "annopath", "imgpath", "imgsetpath". 

```#    with open(imagesetfile, 'r') as f:
#        lines = f.readlines()
#    imagenames = [x.strip() for x in lines]
    imagenames = []
    for file in os.listdir(osp.join(rootpath,imagesetfile)):
        image = os.fsdecode(file)
        if image.endswith('.jpg'):
            imagenames.append(image[:-4])```

And I have changed this according to the dataset. You can use print statement. I think the problem is it is not able to read images and giving zero for all classes.

**Comment by AnivChakravarty** [comment]
Also, be sure to clear your annotations cache upon making any changes to your ground truth format
